src/tools/add_slice_to_custom_type.ts

```diff
@@ -0,0 +1,210 @@
+			} catch (error) {
+				// noop, we don't wanna block the tool call if the tracking fails
+				if (process.env.DEBUG) {
+					console.error(
+						"Error while tracking 'add_slice_to_custom_type' tool call",
+						error,
+					)
+				}
+			}
+
+			// Read the slice model
+			const sliceModelAbsolutePath = joinPath(
+				sliceDirectoryAbsolutePath,
+				"model.json",
+			)
+
+			let parsedSliceModel: SharedSlice
+			try {
+				parsedSliceModel = JSON.parse(
+					readFileSync(sliceModelAbsolutePath, "utf-8"),
+				)
+			} catch (error) {
+				throw new Error(
+					`Invalid JSON format for slice model at ${sliceModelAbsolutePath}: ${getErrorMessage(error)}`,
+				)
+			}
+
+			// Read the custom type model
+			const customTypeModelAbsolutePath = joinPath(
+				customTypeDirectoryAbsolutePath,
+				"index.json",
+			)
+
+			let customTypeParsedModel: StaticCustomType
+			try {
+				customTypeParsedModel = JSON.parse(
+					readFileSync(customTypeModelAbsolutePath, "utf-8"),
+				)
+			} catch (error) {
+				throw new Error(
+					`Invalid JSON format for page/custom type model at ${customTypeModelAbsolutePath}: ${getErrorMessage(error)}`,
+				)
+			}
+
+			// Add the slice to the first slice field in the model
+			let sliceAlreadyAdded = false
+
+			const hasSliceZone = Object.entries(customTypeParsedModel.json).some(
+				([sectionName, fields]) => {
+					return Object.entries(fields).some(([fieldId, field]) => {
+						if (field?.type === "Slices") {
+							if (
+								Object.keys(field.config?.choices || {}).includes(
+									parsedSliceModel.id,
+								)
+							) {
+								sliceAlreadyAdded = true
+							} else {
+								const sliceField = customTypeParsedModel.json[sectionName][
+									fieldId
+								] as DynamicSlices
+
+								if (!sliceField.config) {
+									sliceField.config = {}
+								}
+								if (!sliceField.config.choices) {
+									sliceField.config.choices = {}
+								}
+
+								sliceField.config.choices[parsedSliceModel.id] = {
+									type: parsedSliceModel.type,
+								}
+							}
+
+							return true
+						}
+
+						return false
+					})
+				},
+			)
+
+			if (sliceAlreadyAdded) {
+				return {
+					content: [
+						{
+							type: "text",
+							text: `The "${parsedSliceModel.id}" slice already exists in the "${customTypeParsedModel.id}" custom type`,
+						},
+					],
+				}
+			}
+
+			// If the custom type does not have a slice zone, add it to the first section
+			if (!hasSliceZone) {
+				const firstSection = Object.keys(customTypeParsedModel.json)[0]
+				;(customTypeParsedModel.json[firstSection].slices as DynamicSlices) = {
+					type: "Slices",
+					fieldset: "Slice Zone",
+					config: {
+						choices: {
+							[parsedSliceModel.id]: {
+								type: parsedSliceModel.type,
+							},
+						},
+					},
+				}
+			}
+
+			const validationResult = CustomType.decode(customTypeParsedModel)
+
+			if (validationResult._tag === "Left") {
+				if (process.env.PRISMIC_DEBUG) {
```

Comment:
Why do you have a console log here but not down next try catch?

When do you want a console log or not?

The throw is not enought for us to debug?